pyccl/halos/profiles/einasto.py

```diff
@@ -47,15 +47,20 @@ def __init__(self, *, concentration, truncated=True, alpha='cosmo',
         self.truncated = truncated
         self.alpha = alpha
         super().__init__(mass_def=mass_def, concentration=concentration)
-        mvir = MassDef("vir", "matter")
-        self._to_virial_mass = mass_translator(
-            mass_in=self.mass_def, mass_out=mvir,
-            concentration=self.concentration)
+        self._init_mass_translator()
         self.update_precision_fftlog(padding_hi_fftlog=1E2,
                                      padding_lo_fftlog=1E-2,
                                      n_per_decade=1000,
                                      plaw_fourier=-2.)
 
+    @unlock_instance
+    def _init_mass_translator(self):
+        # Set the mass translator to Mvir as an attribute.
+        # TODO: Move to `__init__` in CCLv3.
+        self._to_virial_mass = mass_translator(
+            mass_in=self.mass_def, mass_out=MassDef("vir", "matter"),
+            concentration=self.concentration)
+
```

Comment:
Why couldn't this stay where it was?

Reply:
it was in `__init__`, however we need to update the translator's `mass_def` every time users pass a new one through `real`, `fourier` etc.